### ISSUE TITLE
Crash if MMDVMHost shuts down before DMRGateway

### ISSUE THREAD

**Issue by dg9vh** [g4klx/DMRGateway#7]
If the MMDVMHost-process fails or shuts down, the DMRGateway crashes with following error:

DMRGateway: DMRNetwork.cpp:55: CDMRNetwork::CDMRNetwork(const string&, unsigned int, unsigned int, unsigned int, const string&, const char*, bool): Assertion `id > 1000U' failed.

Happens with actual commit c2bcec4a59a5b28f142a55d0b2119818d7eb5b96 and several before...

**Comment by g0wfv** [comment]
I've pinned the circumstances to recreate this one down.

If you have an already running instance of DMRGateway and MMDVMHost then restart the DMRGateway with MMDVMHost still running, the next time you restart MMDVMHost, DMRGateway bombs out as follows ....

```
I: 2017-06-02 10:00:48.393 This software is for use on amateur radio networks only,
I: 2017-06-02 10:00:48.397 it is to be used for educational purposes only. Its use on
I: 2017-06-02 10:00:48.400 commercial networks is strictly prohibited.
I: 2017-06-02 10:00:48.401 Copyright(C) 2017 by Jonathan Naylor, G4KLX and others
M: 2017-06-02 10:00:48.402 DMRGateway-20170522 is starting
M: 2017-06-02 10:00:48.403 Built 10:54:06 Jun  2 2017 (GitID #a232add)
I: 2017-06-02 10:00:48.403 MMDVM Network Parameters
I: 2017-06-02 10:00:48.403     Rpt Address: 127.0.0.1
I: 2017-06-02 10:00:48.404     Rpt Port: 62032
I: 2017-06-02 10:00:48.404     Local Address: 127.0.0.1
I: 2017-06-02 10:00:48.404     Local Port: 62031
M: 2017-06-02 10:00:48.405 DMR, Opening MMDVM Network
M: 2017-06-02 10:00:48.407 Waiting for MMDVM to connect.....
M: 2017-06-02 10:00:53.941 MMDVM has connected
I: 2017-06-02 10:00:53.942 DMR Network 1 Parameters
I: 2017-06-02 10:00:53.942     Id: 0
I: 2017-06-02 10:00:53.942     Address: bm-dmr.uk
I: 2017-06-02 10:00:53.942     Port: 62031
I: 2017-06-02 10:00:53.942     Local: random
DMRGateway: DMRNetwork.cpp:55: CDMRNetwork::CDMRNetwork(const string&, unsigned int, unsigned int, unsigned int, const string&, const char*, bool): Assertion `id > 1000U' failed.
Aborted
```

So, the sequence to reproduce is ...

Start DMRgateway
Start MMDVMHost

Stop DMRGateway
Start DMRGateway

Stop MMDVMHost  (before the network times out and reconnects!)

**Comment by g0wfv** [comment]
Further to this, it would seem the disconnecting MMDVMHost process although not connected to the DMRGateway at this stage, but it thinks it is, is maybe sending something down the UDP pipe on close which DMRGateway is misinterpreting as a connected host, however it hasn't received an ID from a connecting host, so the assertion that an ID must me greater than 1000 bombs the gateway out.

Fixed and PR submitted #16
